### PR TITLE
[spiral/router] Opening of `http.request` scope added

### PR DESCRIPTION
**Pull request by msmakouz**
| Q             | A
| ------------- | ---
| Bugfix?       | ❌
| Breaks BC?    | ❌ 
| New feature?  | ✔️

## What was changed

Opening of `http.request` scope added, `Proxy` attributes added to get dependencies in the correct scope.

### CODE REVIEW

File: src/AnnotatedRoutes/tests/IntegrationTest.php [hunk 1]
```diff
@@ -85,8 +91,10 @@ public function post(
         array $headers = [],
         array $cookies = []
     ): ResponseInterface {
-        return $this->app->getHttp()->handle(
-            $this->request($uri, 'POST', [], $headers, $cookies)->withParsedBody($data)
+        return $this->app->getContainer()->runScope(
+            new Scope(ScopeName::Http),
```
Comment (wolfy-j) [inline src/AnnotatedRoutes/tests/IntegrationTest.php:95]: we will come up with better name for the enum before release

File: src/AnnotatedRoutes/tests/IntegrationTest.php [hunk 2]
```diff
@@ -63,7 +66,10 @@ public function get(
         array $headers = [],
         array $cookies = []
     ): ResponseInterface {
-        return $this->app->getHttp()->handle($this->request($uri, 'GET', $query, $headers, $cookies));
+        return $this->app->getContainer()->runScope(
+            new Scope(ScopeName::Http),
+            fn (Http $http) => $http->handle($this->request($uri, 'GET', $query, $headers, $cookies))
```
Comment (wolfy-j) [inline src/AnnotatedRoutes/tests/IntegrationTest.php:71]: i wonder if we can have container in HTTP scope already here... @roxblnfk like $this->getHttpScopeContainer() and then we work from there. can we do that?

File: src/Hmvc/tests/CoreTest.php [hunk 3]
```diff
@@ -7,191 +7,232 @@
...
     {
-        $core = new SampleCore(new Container());
-        $this->assertSame('Hello, Antony.', $core->callAction(
-            DummyController::class,
-            'index',
-            ['name' => 'Antony']
-        ));
+        $core = new SampleCore($this->root);
+
+        $this->root->runScope(new Scope(ScopeName::Http), function () use ($core): void {
```
Comment (wolfy-j) [inline src/Hmvc/tests/CoreTest.php:29]: we have to simplify this for users, it's ok in framework, but people won't be creating HTTP scope by themselves.

Comment (wolfy-j) [reply src/Hmvc/tests/CoreTest.php:29]: also, the test can be in TEST scope overall, most likely "HTTP" scope won't be doing much, HTTP.request will.

File: src/Hmvc/tests/CoreTest.php [hunk 4]
```diff
@@ -7,191 +7,232 @@
...
     public function testCallNotController(): void
     {
         $this->expectException(ControllerException::class);
 
-        $core = new SampleCore(new Container());
-        $core->callAction(SampleCore::class, 'index', [
-            'name' => 'Antony',
-        ]);
+        $core = new SampleCore($this->root);
+        $this->root->runScope(new Scope(ScopeName::Http), function () use ($core): void {
```
Comment (wolfy-j) [inline src/Hmvc/tests/CoreTest.php:150]: yeah, too much overhead

File: src/AnnotatedRoutes/tests/IntegrationTest.php (hunk 2 above)
Comment (roxblnfk) [reply src/AnnotatedRoutes/tests/IntegrationTest.php:71]: I think about using attributes for scoped tests:

```php
#[TestScope('http.request')]
public function testSomethingInHttpRequest() {...}
```

File: tests/Framework/Bootloader/Http/HttpBootloaderTest.php [hunk 5]
```diff
@@ -21,7 +24,17 @@ final class HttpBootloaderTest extends BaseTestCase
 {
     public function testHttpBinding(): void
     {
-        $this->assertContainerBoundAsSingleton(Http::class, Http::class);
+        $this->getContainer()->runScope(new Scope(ScopeName::Http), function (Container $container): void {
```
Comment (roxblnfk) [inline tests/Framework/Bootloader/Http/HttpBootloaderTest.php:27]: Looks like you can use TestScope attribute here

Comment (msmakouz) [reply tests/Framework/Bootloader/Http/HttpBootloaderTest.php]: Yes, you're right

File: src/AnnotatedRoutes/tests/IntegrationTest.php (hunk 2 above)
Comment (msmakouz) [reply src/AnnotatedRoutes/tests/IntegrationTest.php]: @wolfy-j @roxblnfk Added **TestScope** attribute to the spiral/testing package, simplified tests.

**Review by roxblnfk — APPROVED** [review]

**Pull request merged**